feature-pack/src/main/resources/modules/system/layers/base/io/smallrye/opentracing/main/module.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2018 Red Hat, Inc.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~   http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<module xmlns="urn:jboss:module:1.8" name="io.smallrye.opentracing">
+    <properties>
+        <property name="jboss.api" value="private"/>
+    </properties>
+
+    <resources>
+        <artifact name="${io.smallrye:smallrye-opentracing}"/>
+    </resources>
+
+    <dependencies>
+        <module name="java.logging"/>
+        <module name="javax.annotation.api"/>
+        <module name="javax.enterprise.api" />
+        <module name="javax.inject.api"/>
+        <module name="javax.ws.rs.api"/>
+
+        <module name="io.opentracing.opentracing-api"/>
+        <module name="io.opentracing.contrib.opentracing-jaxrs2"/>
+
+        <module name="org.eclipse.microprofile.opentracing"/>
+        <module name="org.wildfly.microprofile.opentracing-smallrye"/>
```

Comment:
Why is this dependency needed? `smallrye-opentracing` should not need anything from the wildfly integration module.

Reply:
See the PR description from #11572, but I understand that this (circular) dependency is needed in order for the `i.s.opentracing` module to be able to consume a bean produced by the `o.w.m.opentracing-smallrye` module.

Reply:
I see. Hm, then a circular dependency is probably the only way how to implement this.